romeo/src/state.rs

```diff
@@ -469,13 +473,44 @@ impl State {
 			State::Initialized {
 				deposits,
 				withdrawals,
+				stacks_block_height,
 				..
 			} => {
 				let deposit_tasks = deposits.iter_mut().filter_map(|deposit| {
 					match deposit.mint.as_mut() {
 						None => {
-							deposit.mint = Some(TransactionRequest::Created);
-							Some(Task::CreateMint(deposit.info.clone()))
+							// We often receive the deposit before the
+							// transaction is actually mined. By scheduling the
+							// transaction for a block later than the current
+							// one we make ourselves resilient to mining delays
+							// without complex logic.
+							let scheduled_block_height = *stacks_block_height
+								+ MINT_TRANSACTION_DELAY_BLOCKS;
+							deposit.mint =
+								Some(TransactionRequest::Scheduled {
+									block_height: scheduled_block_height,
+								});
+							info!("Scheduled deposit {} for minting on stacks block height {}.",
+								deposit.info.txid, scheduled_block_height);
+							None
+						}
+						Some(TransactionRequest::Scheduled {
+							block_height,
+						}) => {
+							if stacks_block_height.ge(&block_height) {
+								// Only initiate the mint task if the current
+								// stacks block is or is after the stacks block
+								// for which the mint is scheduled.
+								deposit.mint =
+									Some(TransactionRequest::Created);
+								info!(
```

Comment:
I'd convert this to a `trace!` as it's not too important to see in normal operation, or `debug!`.

Reply:
Done.